src/js/AppHeader.js

```diff
@@ -17,19 +18,52 @@ import ScrollableMessage from './ScrollableMessage';
 
 import {ReactComponent as IconMenu} from '../img/icons/icon-menu.svg'
 import {ReactComponent as IconSettings} from '../img/static/0x49.svg'
+import {ReactComponent as IconFSCConnected} from '../img/static/0x30.svg'
+import {ReactComponent as IconFSCDisconnected} from '../img/static/0x31.svg'
 
 class MainMenuSettings extends React.Component {
     render() {
-        return (<div>
+        return (<div className="settings-menu">
                 <Link to="/settings">
-                    <span className="settings-menu">
+                    <span className="settings-menu-icon">
                         <IconSettings/>
                     </span>
                 </Link>
             </div>);
     }
 }
 
+class BackendConnectionStatus extends React.Component {
+    render() {
+        let statusIcon = FileSystemController.isConnected() ? <IconFSCConnected/>
+            : <IconFSCDisconnected/>;
+        let attemptReconnection = (url) => {
+            window.flags.FileSystemApiUrl = url;
+            localStorage.setItem("FileSystemApiUrl", window.flags.FileSystemApiUrl)
```

Comment:
I would suggest only saving `FileSystemApiUrl` in local storage if the connection is successful, maybe just moving this line into the connect resolve method.

Reply:
Good idea. Changed in https://github.com/smartdevicelink/generic_hmi/pull/481/commits/67dd0c091ee8cfd7ffdfc063eaecd70c8693cdf9